v2/gcs-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/processing/dml/DMLGeneratorTest.java

```diff
@@ -671,8 +673,38 @@ public void bitColumnSql() {
         + " 'kk', LastName = BINARY(FROM_BASE64('YmlsX2NvbA=='))"; */
     String sql =
         DMLGenerator.getDMLStatement(
-            modType, tableName, schema, newValuesJson, keyValuesJson, "+00:00");
+            modType, tableName, schema, newValuesJson, keyValuesJson, "+00:00", null);
 
     assertTrue(sql.contains("LastName = BINARY(FROM_BASE64('YmlsX2NvbA=='))"));
   }
+
+  @Test
+  public void customTransformationMatch() {
+    Schema schema = SessionFileReader.read("src/test/resources/customTransformation.json");
+    String tableName = "Singers";
+    String newValuesString = "{\"FirstName\":\"kk\",\"LastName\":\"ll\"}";
+    JSONObject newValuesJson = new JSONObject(newValuesString);
+    String keyValueString = "{\"SingerId\":\"999\"}";
+    JSONObject keyValuesJson = new JSONObject(keyValueString);
+    String modType = "INSERT";
+    Map<String, Object> customTransformation = new HashMap<>();
+    customTransformation.put("FullName", "\'kk ll\'");
+    customTransformation.put("SingerId", "1");
+
+    /*The expected sql is:
+    "INSERT INTO Singers(SingerId,FullName) VALUES (1,'kk ll') ON DUPLICATE KEY"
+        + " UPDATE  FullName = 'kk ll'";*/
+    String sql =
+        DMLGenerator.getDMLStatement(
+            modType,
+            tableName,
+            schema,
+            newValuesJson,
+            keyValuesJson,
+            "+00:00",
+            customTransformation);
+
+    assertTrue(sql.contains("FullName = 'kk ll'"));
+    assertTrue(sql.contains("VALUES (1,'kk ll')"));
```

Comment:
The order of columns is not guaranteed, it often fails in the presubmit. Instead you can better check only the column name = value string

Reply:
Looks good overall, just one comment

Reply:
ignore, it will work for single PK